mlir/lib/Dialect/MIGraphX/IR/MIGraphX.cpp

```diff
@@ -251,6 +251,8 @@ RankedTensorType MIXRShapedType::asMemoryLayoutTensor() const {
 
 RankedTensorType MIXRShapedType::asFlatMemoryTensor() const {
   RankedTensorType memoryTensorType = asMemoryLayoutTensor();
+  if (!memoryTensorType)
+    return nullptr;
```

Comment:
are there any unit-tests for such kind of methods/functions ?

Reply:
Nope

I'd've added a `not --crash` test, if it weren't for the fact that we're still going to fail the example that triggered the crash.

Reply:
That is to say, we could have a test for this invalid input producing an error, but it's trickier (given that, in general, testing via `lit` is preferred over unit tests) it's harder to express "fails in this specific way but doesn't crash" (probably due to LLVM having a propensity to `report_fatal_error`)